AK/JsonParser.cpp

```diff
@@ -120,25 +81,18 @@ String JsonParser::consume_quoted_string()
             final_sb.append('\f');
             break;
         case 'u': {
-            StringBuilder sb;
-            sb.append(consume());
-            sb.append(consume());
-            sb.append(consume());
-            sb.append(consume());
-
-            auto codepoint = AK::StringUtils::convert_to_uint_from_hex(sb.to_string());
-            if (codepoint.has_value()) {
+            auto codepoint = AK::StringUtils::convert_to_uint_from_hex(consume(4));
+            if (codepoint.has_value())
                 final_sb.append_codepoint(codepoint.value());
-            } else {
+            else
                 final_sb.append('?');
-            }
         } break;
         default:
             final_sb.append(escaped_ch);
             break;
         }
     }
-    if (!consume_specific('"'))
+    if (consume() != '"')
```

Comment:
You have a `consume_if_next_is(char)` method, and it looks like it "replaces" `consume_specific`. Is there a reason why you're not using it there - different behaviour, etc... ?
```suggestion
    if (!consume_if_next_is('"'))
```

Reply:
Absolutely, the `consume_if_next_is()` method consumes only if the argument matches with what comes next in the Lexer. But the original `JsonParser::consume_specific()` actually consumed first, and then tested for a match. I thought I would leave this logic be, even though it makes no difference in the end. I'll change it in the next commit.